src/main/java/com/ldtteam/structures/helpers/BlueprintIterator.java

```diff
@@ -0,0 +1,234 @@
+    {
+        return iterateWithCondition(skipCondition, this::increment);
+    }
+
+    /**
+     * Decrement the structure with a certain skipCondition (jump over blocks that fulfill skipCondition).
+     * @param skipCondition the skipCondition.
+     * @return Result of decrement.
+     */
+    public Result decrement(final TriPredicate<BlueprintPositionInfo, BlockPos, World> skipCondition)
+    {
+        return iterateWithCondition(skipCondition, this::decrement);
+    }
+
+    /**
+     * Execute a supplier function to avoid duplicate code for increment and decrement functions.
+     * @param skipCondition the skipCondition.
+     * @param function the supplier function.
+     * @return the Result.
+     */
+    private Result iterateWithCondition(final TriPredicate<BlueprintPositionInfo, BlockPos, World> skipCondition, final Supplier<Result> function)
+    {
+        int count = 0;
+        do
+        {
+            final BlockPos worldPos = structureHandler.getWorldPos().subtract(structureHandler.getBluePrint().getPrimaryBlockOffset()).add(progressPos);
+            final BlueprintPositionInfo info = getBluePrintPositionInfo();
+
+            if(function.get() == Result.AT_END)
+            {
+                return Result.AT_END;
+            }
+            else if (BlockUtils.areBlockStatesEqual(info.getBlockInfo().getState(), structureHandler.getWorld().getBlockState(worldPos), structureHandler::replaceWithSolidBlock, structureHandler.fancyPlacement()))
+            {
+                structureHandler.triggerSuccess(progressPos);
+            }
+            else if (skipCondition.test(info, worldPos, structureHandler.getWorld()))
+            {
+                continue;
+            }
+            return Result.NEW_BLOCK;
+        }
+        while (count++ < structureHandler.getMaxBlocksCheckedPerCall());
+
+        return Result.CONFIG_LIMIT;
+    }
+
+    /**
+     * Change the current progressPos. Used when loading progress.
+     *
+     * @param localPosition new progressPos.
+     */
+    public void setProgressPos(@NotNull final BlockPos localPosition)
+    {
+        this.progressPos.setPos(localPosition.getX() % size.getX(),
+          localPosition.getY() % size.getY(),
+          localPosition.getZ() % size.getZ());
+    }
+
+    /**
+     * Get the blueprint info from the position.
+     * @return the info object.
+     */
+    @NotNull
+    public BlueprintPositionInfo getBluePrintPositionInfo()
+    {
+        return structureHandler.getBluePrint().getBluePrintPositionInfo(progressPos, includeEntities);
+    }
+
+    /**
+     * Set the iterator to include entities.
+     */
+    public void includeEntities()
+    {
+        this.includeEntities = true;
+    }
+
+    /**
+     * Reset the progressPos.
+     */
+    public void reset()
+    {
+        BlockPosUtil.set(this.progressPos, NULL_POS);
+        includeEntities = false;
```

Comment:
I like to turn it on manually only. Phases that want it, turn it on manually, the other phases just don't.

Reply:
ye but we reset on many places, so it may get lost. when using it I'd expect the flag to not get set back to default all the time when its reseting some stuff

Reply:
If it resets, this means automatically that this iteration finished. That's the only moment it is reset.